### PR TITLE
[WIP]: Upgrade automated tests [Old: Do not merge]

### PR DESCRIPTION
**Pull request by Xenwar**
This PR is to be replace by other PRs that are specific to each use case.
Developers choose a use case and create PR for it.

### CODE REVIEW

File: upgrade_tests/osimage_upgrade.sh [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+      hostSelector: {}
+      image:
+        checksum: http://172.22.0.1/images/bionic-server-cloudimg-amd64.img.md5sum
+        url: https://cloud-images.ubuntu.com/bionic/current/bionic-server-cloudimg-amd64.img
+" > "${OUTPUT_FILE}"
+}
+
+echo "Provisioning a control plane node ...."
+#./scripts/v1alphaX/provision_cluster.sh
+#./scripts/v1alphaX/provision_controlplane.sh
```
Comment (Jaakko-Os) [inline upgrade_tests/osimage_upgrade.sh]: uncomment?
add:
echo "Provisioning a cluster"

File: upgrade_tests/osimage_upgrade.sh [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+#if [[ "$?" == '0' ]]; then	  
+#  echo "Successfully upgrade image of a control plane node"
+#  break
+#fi
+
+
+# ------ Alternative options
+# Patch in place |- does not seem to be possible
+# apply a modified file |- does not seem to work due to revision mismtach
+#
```
Comment (Jaakko-Os) [inline upgrade_tests/osimage_upgrade.sh]: uncomment or remove

Comment (Xenwar) [reply upgrade_tests/osimage_upgrade.sh]: Thanks, those commented ones are breaking the test. 
Working on it.

File: upgrade_tests/CP_upgrade/OSImage_upgrade.sh [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+generate_metal3MachineTemplate test1-new-image "${CLUSTER_UID}" "${Metal3MachineTemplate_OUTPUT_FILE}"
+kubectl apply -f "${Metal3MachineTemplate_OUTPUT_FILE}"
+
+# Change metal3MatchineTemplate reference
+kubectl get kubeadmcontrolplane -n metal3 test1-controlplane -o yaml | grep 'spec:' -A 10000 > /tmp/patch-kubeadmcontrolplane.yaml
+sed -i 's/name: test1-controlplane/name: test1-new-image/g' /tmp/patch-kubeadmcontrolplane.yaml
+kubectl patch kubeadmcontrolplane test1-controlplane -n metal3 --type merge --patch "$(cat /tmp/patch-kubeadmcontrolplane.yaml)"
+
+echo "Waiting for upgrade process to complete ...."
+NEW_NODE_NAME=$(kubectl get bmh -n metal3 | grep -v node-3 | grep 'prov' | awk '{{print $1}}')
```
Comment (Jaakko-Os) [inline upgrade_tests/CP_upgrade/OSImage_upgrade.sh:78]: use
"${CHOSEN_NODE}"
instead of 'node-3'

**Comment by Xenwar** [comment]
This PR is no longer needed.
I has been broken into multiple smaller tests based on use cases.

**Pull request closed without merging**